### PR TITLE
Svdesu/ch144/create a circuit design

### PR DESCRIPTION
**Pull request by SVD6**
Please Note: We can go through the code later for efficiency and stuff, it's not that big of a deal what's there works and it's pretty clean. I think the state handling could be a bit better but overall it's not too bad trust. Let me know if there's anything obviously wrong/dumb I'll fix it.

The sounds being used are temporary. We need something much better like a whistle blow or something, we can look into it after we do the beta launch at least. 

There are no known issues until someone finds one lol, do some heavy testing before approving this merge.

### CODE REVIEW

File: app/src/main/java/com/example/chrono/main/circuit/CircuitCreate.kt [hunk 1]
```diff
@@ -88,7 +92,7 @@ class CircuitCreate : BaseActivity() {
     }
 
     private fun validateInputs(): Boolean {
-        if (bind!!.nameInput.isEmpty() or (bind!!.nameInput.editText!!.text.toString() == "")) {
+        if ((bind!!.nameInput.text.toString() == "") or (bind!!.nameInput.text.toString() == "")) {
```
Comment (harigovind1998) [inline app/src/main/java/com/example/chrono/main/circuit/CircuitCreate.kt]: These 2 if conditions are the same

File: app/src/main/res/drawable/ic_close_white.xml [hunk 2]
```diff
@@ -0,0 +1,18 @@
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
+    android:width="40dp"
+    android:height="40dp"
+    android:viewportWidth="40"
```
Comment (harigovind1998) [inline app/src/main/res/drawable/ic_close_white.xml:4]: Doesn't this throw an IDE warning without the dp?

File: app/src/main/res/layout/activity_circuit_timer.xml [hunk 3]
```diff
@@ -3,150 +3,151 @@
     xmlns:app="http://schemas.android.com/apk/res-auto">
 
     <RelativeLayout
+        android:id="@+id/main_layout"
         android:layout_width="match_parent"
         android:layout_height="match_parent">
 
         <ImageView
             android:id="@+id/close_button"
```
Comment (harigovind1998) [inline app/src/main/res/layout/activity_circuit_timer.xml:11]: This button is too close to the top of the screen. Needs to be moved down.

**Review by harigovind1998 — COMMENTED** [review]
loadTimer() needs to be called each time the start button is pressed to allow users to rerun circuit. RN the currentSet is 0 when the circuit is completed and when start is pressed, it only runs through the getReady state and ends.

File: app/src/main/java/com/example/chrono/main/circuit/CircuitCreate.kt (hunk 1 above)
Comment (SVD6) [reply app/src/main/java/com/example/chrono/main/circuit/CircuitCreate.kt]: rip good point

File: app/src/main/res/drawable/ic_close_white.xml (hunk 2 above)
Comment (SVD6) [reply app/src/main/res/drawable/ic_close_white.xml:4]: No it doesn't. I just exported the SVG from figma and imported it here didn't do anything else.

File: app/src/main/res/layout/activity_circuit_timer.xml (hunk 3 above)
Comment (SVD6) [reply app/src/main/res/layout/activity_circuit_timer.xml:11]: fixed

**Pull request merged**